service/surf/vendor/search/tests/tests_elastic_search.py

```diff
@@ -172,6 +172,26 @@ def test_drilldown_search(self):
                 self.assertTrue(item['external_id'])
                 self.assertIsNotNone(item['count'])
 
+    def test_drilldown_with_filters(self):
+        search = self.instance.search(
+            ["biologie"],
+            filters=[
+                {"external_id": "lom.technical.format", "items": ["text"]}
+            ],
+            drilldown_names=['about.repository', 'lom.educational.context', 'lom.technical.format']
+        )
+
+        drilldowns = search['drilldowns']
+        drilldowns_for_format = next((d for d in drilldowns if d['external_id'] == 'lom.technical.format'), None)
+        drilldowns_for_repo = next((d for d in drilldowns if d['external_id'] == 'about.repository'), None)
+
+        total_for_format = sum(item['count'] for item in drilldowns_for_format['items'])
+        total_for_repo = sum(item['count'] for item in drilldowns_for_repo['items'])
+
+        # The counts for format do not include the filter (as it is applied to format)
+        # The counts for repo DO include the format filter, so it returns less results
```

Comment:
Hoe wordt de volgorde bepaald? Kan me ook voorstellen dat repo wel format bevat en format niet repo. Is dat op basis van de volgorde van drilldown names? Zo ja, hoe ordenen we drilldown names in frontend?

Reply:
Ik weet niet helemaal wat je bedoelt. Maar de volgorde zou niet belangrijk moeten zijn, want het is altijd AND tussen categorieën.

Dus stel je kiest als format 'Tekst'. Dan krijg je binnen repo nog alleen de repo's met de counts van Tekst.
Alleen binnen format krijg je dan nog alle counts terug (dus Tekst wordt daar genegeerd).

Andersom als je als Repo Wikiwijs kiest dan krijg je bij format alleen de formats die uit Wikiwijs komen, maar bij repo's krijg je alle repo's (want Wikiwijs wordt daar genegeerd).

En als je zowel 'Tekst' als 'Wikiwijs' selecteert dan krijg je bij formats alle formats die in Wikiwijs zitten en bij repo's alle repo's die Tekst hebben.

Enzovoort...

Hoop dat het duidelijk is 😛 .

Reply:
Ja, ik snap hem nu wel. Is het nog handig om de case met twee filters toe te voegen?

Reply:
Ja, alleen durf ik niet vertrouwen op de counts vanuit de test elastic. Ik heb nu een (mega 😛 ) e2e-test geschreven, die kunnen we ook gewoon in de CI draaien.